doc/source/development.rst

```diff
@@ -216,16 +216,28 @@ Format and Linting
 
 **Running linter locally:** To run the Python linter on a specific file, run
 something like ``flake8 ray/python/ray/worker.py``. You may need to first run
-``pip install flake8``.
+``pip install flake8==3.7.7``.
 
-**Autoformatting code**. We use `yapf <https://github.com/google/yapf>`_ for
+**Autoformatting Python code**. We use `yapf <https://github.com/google/yapf>`_ for
 linting, and the config file is located at ``.style.yapf``. We recommend
-running ``scripts/yapf.sh`` prior to pushing to format changed files.
-Note that some projects such as dataframes and rllib are currently excluded.
+running ``scripts/format.sh`` prior to pushing to format changed files.
+You need to first run ``pip install yapf==0.23.0``,
+``pip install flake8==3.7.7`` and ``pip install flake8-quotes`` to install
+the required Python linters.
 
 **Running CI linter:** The Travis CI linter script has multiple components to
-run. We recommend running ``ci/travis/format.sh``, which contains both linter
-for python and C++ codes. In addition, there are other formatting checkers for
+run. We recommend running ``scripts/format.sh``, which contains both linter
+for python and C++ codes.
+
+.. note::
+
+    You need to install ``clang-format`` version 7.0.0 to run the C++ linting.
+    On Ubuntu you can install it with ``sudo apt install clang-format-7.0`` and
+    on macOS you can follow
+    `this guide <https://gist.github.com/ffeu/0460bb1349fa7e4ab4c459a6192cbb25#gistcomment-2765744>`_.
```

Comment:
I just tried these instructions for MacOS and they didn't work.

```
(base) rkn@Roberts-MacBook-Pro ray % curl https://gist.githubusercontent.com/ffeu/0460bb1349fa7e4ab4c459a6192cbb25/raw/4ac5c1aef6d24849b96a0d36b6417798289722fe/clang-format@7.rb -o $(brew --repo)/Library/Taps/homebrew/homebrew-core/Formula/clang-format@7.rb

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2446  100  2446    0     0   8348      0 --:--:-- --:--:-- --:--:--  8348
(base) rkn@Roberts-MacBook-Pro ray % brew install clang-format@7

Updating Homebrew...
To restore the stashed changes to /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core run:
  'cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core && git stash pop'
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> New Formulae
corral              fastlane            imgproxy            powerline-go
crc                 ffuf                krew                pympress
detekt              findent             libffcall           python@3.8
diesel              flint               mavsdk              skymaker
docker-slim         git-remote-gcrypt   meilisearch         terrahelp
dune                gnirehtet           node-sass           texlab
ensmallen           hpack               okteto              uni
==> Updated Formulae
gnutls ✔                                 libpq
unbound ✔                                libqalculate
abcmidi                                  librdkafka
activemq                                 libsixel
aliyun-cli                               libsoup
allure                                   libssh
angular-cli                              libtorrent-rasterbar
ansible                                  libvpx
ansible-lint                             linkerd
apktool                                  lmod
appscale-tools                           logtalk
arduino-cli                              makensis
armadillo                                mariadb-connector-c
asdf                                     mesa
asio                                     mesos
ask-cli                                  micronaut
aspectj                                  minikube
atlantis                                 minizip2
audacious                                mitmproxy
aws-cdk                                  mockserver
aws-elasticbeanstalk                     mono
aws-sdk-cpp                              monolith
awscli                                   mosquitto
awsume                                   msgpack
azure-cli                                mutt
babel                                    nano
ballerina                                navi
basex                                    ncmpc
bash-completion@2                        nco
bashdb                                   ncview
batik                                    neomutt
bazel                                    netdata
bazelisk                                 nethack
bcftools                                 netlify-cli
bedtools                                 newlisp
berglas                                  newman
bind                                     nginx
bison                                    ngt
bit                                      nim
bitrise                                  nlohmann-json
bitwise                                  nnn
bluepill                                 node
borgmatic                                node@10
brew-php-switcher                        node@12
bumpversion                              node@8
bundletool                               nomad
bup                                      norm
c-blosc                                  noti
calicoctl                                notmuch
cargo-c                                  now-cli
cargo-completion                         nsd
cash-cli                                 nspr
cassandra@2.1                            nss
cassandra@2.2                            numpy@1.16
certbot                                  nushell
cfn-lint                                 nvm
checkstyle                               ocaml-num
cimg                                     ocrmypdf
citus                                    octomap
ckan                                     omega
cln                                      oniguruma
cmake                                    onioncat
conan                                    onscripter
contentful-cli                           open-scene-graph
convox                                   openal-soft
coq                                      openapi-generator
crc32c                                   opencascade
crystal                                  opencv@2
csvq                                     openimageio
cups                                     operator-sdk
dartsim                                  orc
dashing                                  ortp
dav1d                                    osc
debianutils                              osm2pgsql
deno                                     osmium-tool
deployer                                 osslsigncode
devspace                                 overmind
dhall                                    packer
dhall-bash                               paket
dhall-json                               pandoc
diff-pdf                                 pandoc-citeproc
diffr                                    paps
django-completion                        partio
dlib                                     passenger
dnscrypt-proxy                           pcapplusplus
docfx                                    pdf-redact-tools
doctl                                    pdftoipe
duck                                     pdns
duplicity                                pdnsrec
dvc                                      petsc
easyengine                               petsc-complex
eccodes                                  pgpool-ii
elm                                      phoronix-test-suite
embulk                                   php
ephemeralpg                              php-code-sniffer
erlang                                   php@7.2
eslint                                   php@7.3
ethereum                                 phpstan
exploitdb                                phpunit
faas-cli                                 pnpm
faust                                    ponyc
ffsend                                   poppler
file-formula                             protobuf
fio                                      pspg
firebase-cli                             pulumi
flow                                     pumba
fluxctl                                  pushpin
fmt                                      py2cairo
fonttools                                pyinvoke
freeciv                                  qalculate-gtk
fribidi                                  qbs
frotz                                    qemu
fuseki                                   qpid-proton
futhark                                  qt
fwup                                     quantlib
fx                                       r
fzf                                      rabbitmq
gatsby-cli                               rabbitmq-c
gauche                                   rargs
geogram                                  rav1e
geographiclib                            rbspy
get_iplayer                              re2
ghq                                      recon-ng
gimme-aws-creds                          remctl
git                                      rethinkdb
git-annex                                rex
git-delta                                riemann
git-extras                               rsnapshot
git-plus                                 ruby-build
gitleaks                                 rust
gitmoji                                  rustup-init
gitversion                               samtools
gleam                                    sbt
glew                                     sceptre
glib                                     scrcpy
glib-networking                          sdb
glooctl                                  serverless
gloox                                    simutrans
gmic                                     singular
gmsh                                     skaffold
gnome-autoar                             snakemake
gnupg                                    snapcraft
gnuplot                                  soci
gnuradio                                 solr
go                                       sonobuoy
go@1.12                                  sops
godep                                    spin
golang-migrate                           sqlcipher
google-authenticator-libpam              sqlmap
goreman                                  srt
grakn                                    ssh-copy-id
groonga                                  ssh-vault
grpc                                     sslyze
gst-editing-services                     sstp-client
gst-libav                                starship
gst-plugins-bad                          stress-ng
gst-plugins-base                         subliminal
gst-plugins-good                         swiftformat
gst-plugins-ugly                         swiftlint
gst-python                               syncthing
gst-rtsp-server                          sysbench
gst-validate                             tcsh
gstreamer                                tektoncd-cli
gtk-mac-integration                      telegraf
gwyddion                                 teleport
h3                                       terraform
haxe                                     terraform-docs
helm                                     terraformer
helmfile                                 terragrunt
helmsman                                 terrahub
hledger                                  tflint
hlint                                    tile38
homeassistant-cli                        tintin
howdoi                                   tmux
hstr                                     tmuxinator
htslib                                   tomcat
hugo                                     tomcat@7
i2p                                      tomcat@8
imagemagick                              tomee-plus
imagemagick@6                            tomee-webprofile
imlib2                                   tor
ios-deploy                               traefik
iozone                                   traefik@1
istioctl                                 typescript
ivy                                      unrar
jadx                                     v8
jboss-forge                              verilator
jenkins                                  vim
jenkins-lts                              vips
jfrog-cli-go                             vulkan-headers
jupyterlab                               weaver
just                                     webpack
k6                                       wiki
kakoune                                  wireguard-tools
khard                                    wireshark
kind                                     wolfssl
kotlin                                   wtf
ktlint                                   wtfutil
kube-aws                                 wxmaxima
kubernetes-cli                           xcodegen
kustomize                                xctool
kyma-cli                                 xmake
landscaper                               xmrig
launch_socket_server                     xrootd
libgit2                                  xvid
liblouis                                 yadm
libmagic                                 yamllint
libosinfo                                yarn
libosmium                                zeek
libphonenumber
==> Deleted Formulae
aap                 closure-linter      gost                nicotine-plus
archivemail         curlish             headphones          php@7.1
boost-python@1.59   cvs2svn             ino                 shocco
borg                ddar                iron-functions      sshconfigfs
bubbros             docker-cloud        logentries          tnote
bzr-fastimport      euca2ools           muttils             warp

Error: No available formula with the name "clang-format@7" 
==> Searching for a previously deleted formula (in the last month)...
Warning: homebrew/core is shallow clone. To get complete history run:
  git -C "$(brew --repo homebrew/core)" fetch --unshallow

Error: No previously deleted formula found.
==> Searching for similarly named formulae...
Error: No similarly named formulae found.
==> Searching taps...
==> Searching taps on GitHub...
Error: No formulae found in taps.
(base) rkn@Roberts-MacBook-Pro ray % clang-format --version

zsh: command not found: clang-format
```

Reply:
Interesting, for me it worked. What's your brew version?

```
(base) ➜  python git:(master) ✗ brew --version
Homebrew 2.2.1
Homebrew/homebrew-core (git revision bf9e; last commit 2019-12-22)
Homebrew/homebrew-cask (git revision 2e053; last commit 2019-12-22)
```